src/main/scala/workflow/graph/PipelineEnv.scala

```diff
@@ -30,6 +30,14 @@ class PipelineEnv {
   def setOptimizer(optimizer: Optimizer): Unit = {
     _optimizer = optimizer
   }
+
+  /**
+   * Reset this PipelineEnv (clear state and set the Optimizer to the DefaultOptimizer)
+   */
+  def reset(): Unit = {
+    state.clear()
+    setOptimizer(DefaultOptimizer)
```

Comment:
We should make this method `private[keystone]` or something like that where its visible to internal classes only (i.e. to avoid users calling this). @etrain you can probably do this in your flip-the-switch PR